permitpool/permitpool.go

```diff
@@ -0,0 +1,38 @@
+package permitpool
```

Comment:
Can you add a package doc to describe this package? It might also be helpful to provide an example or two of how this might be used, either in the package doc, or with run-able examples.

Reply:
Added a package doc and a runnable example, thanks!